octobot_trading/exchanges/exchange_manager.py

```diff
@@ -94,6 +95,7 @@ def __init__(self, config, exchange_class_string):
         self.exchange_personal_data = None
         self.exchange_symbols_data = None
         self.trader = None
+        self.trading_modes = []
```

Comment:
We can maybe found a usage for this attribute : 
- In `exchange_manager.stop()` : iterate on the list to call trading mode `stop` method (that should stop consumer)

Reply:
We could, it would be a cleaner way than stopping them with all the exchange channels but it's not necessary at the moment, do you think we should add it ? Having them available somewhere will also be useful in the future for trading modes that will require interactions with other trading modes (multi symbol stuff / arbitrage)

Reply:
I think it's necessary to stop them when their `exchange_manager` is stopping.

> Having them available somewhere will also be useful in the future for trading modes that will require interactions with other trading modes (multi symbol stuff / arbitrage)

Yes I don't say we shouldn't add this attribute. I am trying to find some usage for it.

Reply:
Right, we need to stop them with their ```exchange_manager```, it's already done in ```stop_exchange_channels``` called in ```stop``` but this is only valid if a trading mode is registered as an exchange channel (which should always be the case but is not forced). Therefore I think you are right, we can call a stop for exchange managers there

Reply:
I didn't check before but ```AbstractTradingMode``` doesn't have a ```stop``` method nor a consumer/producer list. What do you think about adding them in ```AbstractTentacle``` and fill them with the returned list from ```create_producers``` and ```create_consumers``` ?

Reply:
Why do you want to add a `stop` method in `AbstractTentacle` ? Why not just in `AbstractTradingMode` ?

Reply:
Wrong copy-paste, I meant ```AbstractTradingMode```

Reply:
Ok :+1: